api/institutions/serializers.py

```diff
@@ -201,13 +201,38 @@ def get_absolute_url(self, obj):
             },
         )
 
+class UniqueDeptIDField(IDField):
+    """Creates a unique department ID of the form "<institution-id>-<dept-id>"."""
+
+    def __init__(self, *args, **kwargs):
+        kwargs['source'] = kwargs.pop('source', 'name')
+        kwargs['help_text'] = kwargs.get('help_text', 'Unique ID that is a compound of two objects. Has the form "<institution-id>-<dept-id>". Example: "cos-psych"')
+        super(UniqueDeptIDField, self).__init__(*args, **kwargs)
+
+    def _get_resource_id(self):
+        return self.context['request'].parser_context['kwargs']['institution_id']
+
+    # override IDField
+    def get_id(self, obj):
+        resource_id = self._get_resource_id()
+        related_id = obj._id.replace(' ', '-')
+        if related_id.startswith(f'{resource_id}-'):
+            return related_id
+        return f'{resource_id}-{related_id}'
+
+    def to_representation(self, value):
+        resource_id = self._get_resource_id()
+        related_id = super(UniqueDeptIDField, self).to_representation(value).replace(' ', '-')
+        if related_id.startswith(f'{resource_id}-'):
```

Comment:
Ok to be explicitly clear about this when I said "having it could be problematic if a department is named something like 'cos-Department Name' the id will be 'cos-cos-Department Name' I ONLY meant that applied to the absolute first PR you submitted for this issue. Sorry if I was confusing about this, but now this could actually break ember. So just remove any special casing for 'cos-cos-Department Name'